lib/onEvent/message.js

```diff
@@ -40,34 +86,38 @@ function checkMessageForCommand (msg, isEdit) {
 	if (isMention(msg.text.split(' '))) {
 		if (Kirbi.Config.elizaEnabled) {
 			//If Eliza AI is enabled, respond to @mention
-			var message = msg.text.replace(`<@${Kirbi.Slack._self.id}> `, '');
+			let message = msg.text.replace(`<@${self.id}> `, '');
 			Kirbi.Slack.sendMessage(Kirbi.Eliza.transform(message), msg.channel);
+			return;
 		} else {
 			Kirbi.Slack.sendMessage('Yes?', msg.channel);
 			return;
 		};
 	};
+	//check for IM
+	if (typeof Kirbi.Slack.dataStore.getDMById(msg.channel) !== 'undefined') {
+		Kirbi.Slack.sendMessage(`I don't respond to direct messages.`, msg.channel);
+		return;
+	};
 	//check if message is a command
 	if (msg.text.startsWith(Kirbi.Config.commandPrefix)) {
 		let allCommands = Object.assign(Kirbi.Commands, Kirbi.slackCommands);
-		var cmdTxt = msg.text.split(' ')[0].substring(Kirbi.Config.commandPrefix.length).toLowerCase();
-		var suffix = msg.text.substring(cmdTxt.length + Kirbi.Config.commandPrefix.length + 1); //add one for the ! and one for the space
-		var cmd = allCommands[cmdTxt];
-		if (msg.type === "dm") {
-			Kirbi.Slack.sendMessage(`I don't respond to direct messages.`, msg.channel);
-		} else if (cmdTxt === 'help') {
-			//help is special since it iterates over the other commands
+		let cmdTxt = msg.text.split(' ')[0].substring(Kirbi.Config.commandPrefix.length).toLowerCase();
+		let suffix = msg.text.substring(cmdTxt.length + Kirbi.Config.commandPrefix.length + 1); //add one for the ! and one for the space
+		let cmd = allCommands[cmdTxt];
+		if (cmdTxt === 'help') {
+			let DM = Kirbi.Slack.dataStore.getDMByUserId(msg.user).id;
 			if (suffix) {
-				var cmds = suffix.split(' ').filter(function (cmd) { return allCommands[cmd] });
-				var info = "";
+				let cmds = suffix.split(' ').filter(function (cmd) { return allCommands[cmd] });
```

Comment:
filter will do a loop, why not make it a forEach and build the info rather than looping through all commands twice?

Reply:
the filter makes sure that the supplied arguments are actually commands.

then we take the list and work through it

Reply:
or we can just do a foreach, see if it is a command and if so do it. This way we are only looping once.